### PR TITLE
Faster XYZMap.get()

### PR DESCRIPTION
**Pull request by Foghrye4**
Do you need a JMH benchmark on this?

### CODE REVIEW

File: src/main/java/cubicchunks/util/XYZMap.java [hunk 1]
```diff
@@ -55,7 +55,8 @@
     /**
      * backing array containing all elements of this map
      */
-    @Nonnull private XYZAddressable[] buckets;
+    @Nonnull private XYZAddressable[] bucketsByPointer;
+    @Nonnull private XYZAddressable[] bucketsByHash;
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/util/XYZMap.java]: add comment to explain why there are 2 separate fields

**Comment by Foghrye4** [comment]
Done.

**Comment by Foghrye4** [comment]
So there is a bug here, as I read at discord?

**Comment by Niko-sk2x** [comment]
Yes, the XYZMap corrupts it's state over time. You may want to add some randomized tests and reproduce it that way, comparing everything against normal HashMap

File: src/main/java/cubicchunks/util/XYZMap.java [hunk 2]
```diff
@@ -179,17 +190,18 @@ public T put(T value) {
         int index = pointers[pointerIndex];
 
         while (index != 0) {
-            XYZAddressable bucket = this.buckets[index];
+            XYZAddressable bucket = this.bucketsByPointer[index];
             if (bucket.getX() == x && bucket.getY() == y && bucket.getZ() == z) {
-                this.buckets[index] = value;
+                this.bucketsByPointer[index] = value;
```
Comment (Foghrye4) [inline src/main/java/cubicchunks/util/XYZMap.java:195]: I forgot to replace old bucketsByHash entry here.

**Pull request merged**